dune_client/api/base.py

```diff
@@ -89,44 +89,47 @@ def default_headers(self) -> Dict[str, str]:
 
     def _build_parameters(
         self,
-        params: Optional[Dict[str, Union[str, int]]] = None,
-        columns: Optional[List[str]] = None,
-        sample_count: Optional[int] = None,
-        filters: Optional[str] = None,
-        sort_by: Optional[List[str]] = None,
-        limit: Optional[int] = None,
-        offset: Optional[int] = None,
         allow_partial_results: str = "true",
+        params: Optional[Dict[str, Any]] = None,
```

Comment:
A good way to start here would be, create the proposed data class and put it here as optional.

See how the function needs to be changed because of the replacement...

Actually, I just noticed this is _build_parameters... This will likely be replaces with a method like `to_dict` on the class itself. The data class can implement both `to_dict` and `from_dict` (this might come for free with the @dataclass decorator although I can't remember). 

Nit: It looks like you forgot to include `allow_partial_results` here, but I really think this `_build_parameters` function will no longer be necessary once this class is in place.